lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4445,6 +4445,13 @@ ActorIsolation ActorIsolationRequest::evaluate(
       if (var->isGlobalStorage() && !isActorType) {
         auto *diagVar = var;
         if (auto *originalVar = var->getOriginalWrappedProperty()) {
+          // temporary 5.10 checking bypass for @TaskLocal <rdar://120907014>
+          // TODO: @TaskLocal should be a macro <rdar://120914014>
+          if (auto *classDecl =
+                  var->getInterfaceType()->getClassOrBoundGenericClass();
+              classDecl && classDecl->getName().is("TaskLocal")) {
```

Comment:
This should preferably check for the exact decl rather than the name -- I'll whip up a snippet for it

Reply:
Like so:
```swift
          if (auto *classDecl =
                  var->getInterfaceType()->getClassOrBoundGenericClass()) {
            auto &ctx = var->getASTContext();
            auto taskLocalDecl = ctx.getTaskLocalDecl();
            if (classDecl == taskLocalDecl) {
              return isolation;
            }
          }
```

That'll avoid triggering for user-defined types of the same name

Reply:
Thanks very much @ktoso I suspected there was a more robust way to check, and I had seen the TaskLocal info in KnownSDKTypes.def but was unsure how it functioned or might help.

Reply:
Happy to help :) 

Usually if we have a type we know about and want to speak about it we put those into Known... and then can on ASTContext get them to compare etc. 👍